CRM/Activity/BAO/Activity.php

```diff
@@ -1783,16 +1783,14 @@ public static function addActivity(
    *   particular component object.
    *
    * @return string
+   * @throws \CRM_Core_Exception
    */
   public static function getActivitySubject($entityObj) {
+    // @todo determine the subject on the appropriate entity rather than from the activity.
     switch ($entityObj->__table) {
       case 'civicrm_membership':
-        $membershipType = CRM_Member_PseudoConstant::membershipType($entityObj->membership_type_id);
-        $subject = $membershipType ? $membershipType : ts('Membership');
-
-        if (is_array($subject)) {
-          $subject = implode(", ", $subject);
-        }
+        $membershipType = CRM_Core_PseudoConstant::getLabel('CRM_Member_BAO_Membership', 'membership_type_id', $entityObj->membership_type_id);
+        $subject = $membershipType ?: ts('Membership');
```

Comment:
should this not be ??

Reply:
This is the same as 
```
if ($membershipType) {
$membershipType;
}
else {
ts('Membership')
}
```

Whereas ?? is 
if (isset($membershipType)) {
$membershipType;
}
else {
ts('Membership')
}